Week5/Solution.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Alyssa Lingad
+ * 
+ * MICP: Week 4
+ */
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import java.util.*;
+
+
+
+public class Solution
+{
+	public static boolean isPalindrome(Node head) throws Exception{
+		if (head == null){
+			throw new Exception();
+		}
+		Node copy = copyLL(head);
```

Comment:
There might be easier ways than cloning the entire input list for this problem. We will discuss in the coming week. I also recommend you code review other students solution (and read up more solutions on internet if possible.).